ext/DataFramesExt.jl

```diff
@@ -0,0 +1,47 @@
+module DataFramesExt
+
+using NEOs
+using NEOs: AbstractAstrometry
+
+import NEOs: RadecMPC, RadarJPL
+
+if isdefined(Base, :get_extension)
+    using DataFrames
+    import DataFrames: DataFrame
+else
+    using ..DataFrames
+    import ..DataFrames: DataFrame
+end
+
+# Method to convert a vector of observations to DataFrame
+function DataFrame(obs::Vector{T}) where {T <: AbstractAstrometry}
+    colnames = fieldnames(T)
+    coltypes = fieldtypes(T)
+    df = DataFrame([name => type[] for (name, type) in zip(colnames, coltypes)])
+    for i in eachindex(obs)
+        push!(df, [getfield(obs[i], colnames[j]) for j in eachindex(colnames)])
+    end 
+
+    return df 
+end 
+
+# Methods to convert a DataFrame to a vector of AbstractAstrometry
+for T in (:(RadecMPC), :(RadarJPL))
+    @eval begin
+        function $T(df::DataFrame)
```

Comment:
This comment is maybe just aesthetics, but while it makes sense to have this defined as some kind of external constructor as you have done, it feels like we should be able to do this via type parameters 🤔... maybe another option would be to define something like

```julia
    function dataframe(x::T) where {T<:AbstractAstrometry}
# ...
    end
```
?

Reply:
Actually, after taking a closer look, it appears as if `RadecMPC` and `RadarJPL` constructors are being defined here (i.e., methods with the name of a type), which then should return an instance of the type they're constructing, but these methods are actually returning `Vector{RadecMPC}` and `Vector{RadarJPL}`, respectively. I think this can be a bit confusing, so I wonder whether we could just define these constructors over a single row of a `DataFrame`, and handle the conversion of the full DataFrame via broadcasting? I mean something like

```julia
   RadecMPC(row_radec) = # constructor a RadecMPC from a single row of a DataFrame
   RadarJPL(row_radar) = # same for RadarJPL
   
   # then, to convert from DataFrame to RadecMPC, RadarJPL:
   RadecMPC.(df_radec)
   RadarJPL.(df_radar)
```

Reply:
(Resolving this conversation in favor of a more concrete suggestion...)